instrumentation/netty/netty-4.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v4_1/BootstrapInstrumentation.java

```diff
@@ -31,35 +37,68 @@
   @Override
   public void transform(TypeTransformer transformer) {
     transformer.applyAdviceToMethod(
-        named("doResolveAndConnect").and(takesArgument(0, SocketAddress.class)),
+        isConstructor().and(takesArguments(0)),
+        BootstrapInstrumentation.class.getName() + "$ConstructorAdvice");
+    transformer.applyAdviceToMethod(
+        named("resolver")
+            .and(takesArguments(1))
+            .and(takesArgument(0, named("io.netty.resolver.AddressResolverGroup"))),
+        BootstrapInstrumentation.class.getName() + "$SetResolverAdvice");
+    transformer.applyAdviceToMethod(
+        named("doConnect")
+            .and(takesArguments(3))
+            .and(takesArgument(0, SocketAddress.class))
+            .and(takesArgument(2, named("io.netty.channel.ChannelPromise"))),
         BootstrapInstrumentation.class.getName() + "$ConnectAdvice");
   }
 
+  @SuppressWarnings("unused")
+  public static class ConstructorAdvice {
+
+    @Advice.OnMethodExit(suppress = Throwable.class)
+    public static void onExit(@Advice.This Bootstrap bootstrap) {
+      // this is already the default value, but we're calling the resolver() method to invoke its
+      // instrumentation
+      bootstrap.resolver(DefaultAddressResolverGroup.INSTANCE);
+    }
```

Comment:
below alternative feels less strange, though I'm not sure what protection we get from muzzle for structure dependency like `FieldValue`? e.g. if the field had a different name for netty 4.1.20, would the CI muzzle check catch that? (or we'd only catch it if we ran tests against that version)
```suggestion
    public static void onExit(
        @Advice.FieldValue(value = "resolver", readOnly = false) AddressResolverGroup<?> resolver) {
      resolver = InstrumentedAddressResolverGroup.wrap(connectionInstrumenter(), resolver);
    }
```

Reply:
No protection at all, I'm afraid; ByteBuddy would just fail when applying this particular advice class and apply everything else.